File: mobile-client/android/UseCase/Project/Detail/LoadProjectDetailUseCase.kt
package app.mobiling.client.usecase.project.detail

import app.mobiling.client.contract.project.detail.ProjectDetailPayload
import app.mobiling.client.data.project.listing.ProjectListingGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class LoadProjectDetailUseCase(
    private val projectListingGateway: ProjectListingGateway,
) {
    suspend operator fun invoke(projectId: String): ProjectDetailPayload =
        projectListingGateway.loadProjectDetail(projectId)
}

File: mobile-client/ios/UseCase/Project/Detail/LoadProjectDetailUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct LoadProjectDetailUseCase {
    let projectListingGateway: ProjectListingGateway

    func callAsFunction(projectId: String) async throws -> ProjectDetailPayload {
        try await projectListingGateway.loadProjectDetail(projectId: projectId)
    }
}
